### DISCUSSION TITLE
New SNES Chinese Repro Board (H01-662k) - Checksum/Dump Changes Every Time

### DISCUSSION THREAD

**Discussion by trgkaigit** [sanni/cartreader#316] [Need help]
I've recently seen a new board revision on some repros, H01-662k instead of the older H10-10k I used to see.  One quirk I've found is these R01-662k boards *always* produce a different checksum when dumping the ROM.  I've observed this on multiple boards, so I don't think it's just a case of a faulty ROM chip.  Either the board design itself is faulty, or there's something funny going on between the cart reader and the chip when performing the dump.

Has anybody else seen these R01-662k boards, and have you seen a similar behavior?

On the bright side, it looks like the point of this revision is adding a 5v<>3.3v voltage regulator (the 662k chip near the bottom right of the battery).  Although I have no idea if that single chip in the location actually solves the problem, but I've also never had any issues after playing around with repros for romhacks and translations for about a decade.

![IMG_1237](https://user-images.githubusercontent.com/92490546/147841568-fb7549ea-1d2f-4424-a9f5-b587969b3baf.jpg)
![IMG_1238](https://user-images.githubusercontent.com/92490546/147841569-f72ceb59-3a55-4a4d-8eac-f9ae520eb0ce.jpg)

**Comment by trgkaigit** [comment]
Okay, after looking at this more closely, the fix was simply changing the Sanni Reader from 5v to 3v.  It now dumps the ROM and the checksum is correct and unchanging.  This does give rise to a new question that I'm hoping a guru can help answer/explain.

My understanding is the "issue" with repro flash was they used 3.3v chips while the SNES was a 5v system, and over enough time (or with a worn out system with bad caps or regulators), this voltage differential could potentially cause damage.  It looks like this board now has a 3.3v<->5v regulator which I assume is to address that potential issue.

If the above is correct, why is the cart reader having issues reading the ROM when set to 5v instead of 3v?  Does the cart reader use a different pin (thus bypassing the 622k regulator) to dump the ROM?

**Reply by Herax-rst** [reply]
Hi,
Very interesting. I can use these sometimes for my personal project.
I also have noticed that the twin PCB used for LoRoms L-02-rst-R10k now uses a AMS1117 5v>3.3v.
So it seems they finally considered not using only leds for volt regulation and use proper components.
Never had issues with it though.

Bad checksum is probably due to overvoltage, that's why you have random results...

I can't properly see the chip attached, is it 29LV640 ?

**Reply by trgkaigit** [reply]
Yep, this is an 29LV640, it's ExLoROM for Fire Emblem: Thracia 776's translation.

I'm curious about the overvoltage though.  Does the SNES read the ROM through a different pin than the Sanni reader's dump utility?  If the Sanni reader is causing bad reads due to overvoltage at 5v, then wouldn't the SNES also have problems since it also runs at 5v?

**Reply by Herax-rst** [reply]
But your board is a H01 board..? Il should only accept HiRoms, not LoRoms ?

I'm not an expert but I tried to flash Near's Bahamut Lagoon translation in a classic H01-10k (ExHirom very respectful of the original hardware) and it was full of bugs, which might be due to overvoltage. Maybe "classic" roms are more flexible about reading errors or maybe we don't see them on screen...

**Reply by Herax-rst** [reply]
Oh and when did you get these ? I made a recent order and got the classic H01-10k but with a M29W640, which is not supported but the cartreader

**Reply by trgkaigit** [reply]
Shoot, sorry, you're right.  This wasn't the Thrcia 776/ExLoROM (been working on that one for a while so it was fresh in my mind).  This was Fire Emblem Genealogy of the Holy War.

I'm actually about to try Bahamut Lagoon with Near's translation.  The bugs are due to Bahamut Lagoon using the SRAM as additional RAM for storing names, so any time the game tries to display a name where you're able to customize it, it starts accessing random parts of memory.  According to what I read, you need to connect SRAM A13 and A14 to snes A16 and A17 (https://www.obscuregamers.com/threads/putting-nears-v1-2-bahamut-lagoon-translation-on-a-cartridge.3810/)

Haven't had a chance to try this out just yet, but it makes sense given how the text corruption seems to be pulling from random parts of memory.

**Reply by trgkaigit** [reply]
> Oh and when did you get these ? I made a recent order and got the classic H01-10k but with a M29W640, which is not supported but the cartreader

I got these about a week ago.  I also had a few LoROM boards that have also been updated with the voltage regulator on them.  They also require bridging WE/WR and disconnecting (cutting) the pads on the far right, similar to the H01-10k boards, which is a bit annoying because I used to be able to just plug a LoROM cart into the reader and replace the ROM on it without opening it up.

**Reply by Herax-rst** [reply]
Good to know ! Well, I tried to link SRAM A13 to SNES A16 and SRAM A114 to SNES A17 and it didin't work...
Glad to have a picture if you succeed in ;)

**Comment by Phobophile89** [comment]
HI,
Sorry for upping this thread,
I have some games on these H01-662K using a MX29LV160. When i'm trying to flash these carts. I get the "Read-only Mode!" error.

So, i discussed with Grok a bit, and found that Leg11 of the MX29LV160 was /WE and that it should connect to Pin 54 of the SNES. So i bridged the "WE" test point with the "WE 1" pad, i made sure the was continuity between LEG11 and Pin 54, but i still get "Read-Only Mode!". The OSCR won't recognise the chip, and if i tell it it's a 2MB HiRom, (For Do-Re-Mi Fantasy) it just again, give me "Read-Only Mode"

Is someone willing to give me some help with that?

**Reply by sanni** [reply]
Enabling write on Chinese repro PCBs is a 2-step process:
1) cut the (very tiny) trace that connects flashrom WE to VCC
2) rewire flashrom WE to SNES WR

In your post you mention having done step 2 but it is unclear if you may have forgotten to do step 1?

In the picture uploaded by trgkaigit this speck of dust would be the trace you need to cut. It connects the two pads together:
<img width="524" height="500" alt="dygtsnj" src="https://github.com/user-attachments/assets/c7257449-7a19-4667-8006-356b56fb70c9" />

**Reply by Phobophile89** [reply]
Thank you SOOO MUCH!

Now i can update my repro carts.

Can't wait for Dragon Quest VI translation to be finished! 👍

**Reply by Chomemel** [reply]
Dragon quest 6 is a 4 mo ROM. With the translation, it becomes a 6 mo if I'm not wrong. 29lv160 is a 2 mo eeprom. DQ6 won't fit. You need a repro with a 29lv640 for it (or 29lv320 if the translated ROM is 4 mo only).

**Reply by Phobophile89** [reply]
Yeah i know!

my 29LV160 are Doremi fantasy
I have a repro dq6 with the latest (unfinished) translation!

**Reply by Phobophile89** [reply]
> Dragon quest 6 is a 4 mo ROM. With the translation, it becomes a 6 mo if I'm not wrong. 29lv160 is a 2 mo eeprom. DQ6 won't fit. You need a repro with a 29lv640 for it (or 29lv320 if the translated ROM is 4 mo only).

So, i just checked, my DQ6 is the NoPrgress version and it is on a MX29LV320EBTI-70G, a 4MB chip. The rom on it is very precisely 4096Kb.

Do you happen to have access to a more recent build? I know some people revived the project and the thread was last updated on Decembre 15th 2025, so it's sure is active!

**Reply by Chomemel** [reply]
Yeah, my bad, it's 4 Mo allright. I did a cartridge with a french translation and thought maybe it's the French version that is 6 Mo, but I just checked and it's 4 Mo too. So 29lv320 is the way to go indeed.
Maybe it's another DQ  which turns to 6 Mo with the translation (DQ3?)
